app/models/miq_provision_virt_workflow.rb

```diff
@@ -335,14 +337,14 @@ def allowed_templates(options = {})
 
       unless tag_conditions.blank?
         _log.info("Filtering VM templates with the following tag_filters: <#{tag_conditions.inspect}>")
-        vms = MiqTemplate.in_my_region.where(condition).find_tags_by_grouping(tag_conditions, :ns => "/managed")
+        templates = templates.where(condition).find_tags_by_grouping(tag_conditions, :ns => "/managed")
       end
     end
 
-    # Only select the colums we need
-    vms = vms.select(:id, :type, :name, :guid, :uid_ems, :ems_id, :cloud_tenant_id)
+    # Only select the columns we need
+    templates = templates.select(:id, :type, :name, :guid, :uid_ems, :ems_id, :cloud_tenant_id)
 
-    allowed_templates_list = source_vm_rbac_filter(vms, condition, VM_OR_TEMPLATE_EXTRA_COLS).to_a
+    allowed_templates_list = source_vm_rbac_filter(templates, condition, VM_OR_TEMPLATE_EXTRA_COLS).to_a
```

Comment:
does this convert the objects to an id and then immediately back into an object?  Or does the VM_OR_TEMPLATE_EXTRA_COLS somehow limit it?

Reply:
I can look at that in a followup.  I was trying to minimize changes for backport of just this bug fix.  There's lots of shiny objects of performance improvements possible here but didn't want to make unrelated changes here.